File: src_f/input_cntl.f
!C
!C***
!C*** INPUT_CNTL
!C***
!C
      subroutine INPUT_CNTL
      use pfem_util

      implicit REAL*8 (A-H,O-Z)

      if (my_rank.eq.0) then
        open (11,file= 'INPUT.DAT', status='unknown')
        read (11,'(a80)') HEADER
        read (11,*) ITER
        read (11,*) COND, QVOL
        read (11,*) RESID
        read (11,*) NFLAG
        close (11)
      endif

      call MPI_BCAST (HEADER, 80, MPI_CHARACTER, 0, MPI_COMM_WORLD,ierr)
      call MPI_BCAST (ITER   , 1, MPI_INTEGER, 0, MPI_COMM_WORLD, ierr)
      call MPI_BCAST (COND   , 1, MPI_DOUBLE_PRECISION, 0, 
     &                                            MPI_COMM_WORLD, ierr)
      call MPI_BCAST (QVOL   , 1, MPI_DOUBLE_PRECISION, 0, 
     &                                            MPI_COMM_WORLD, ierr)
      call MPI_BCAST (RESID  , 1, MPI_DOUBLE_PRECISION, 0, 
     &                                            MPI_COMM_WORLD, ierr)

      pfemRarray(1)= RESID
      pfemIarray(1)= ITER

      return
      end

File: src_c/input_cntl.c
/***
 *** INPUT_CNTL
 ***/
/** PARALLEL VERSION **/
#include <stdio.h>
#include <stdlib.h>
#include "pfem_util.h"
/*** external functions **/
extern void ERROR_EXIT (int, int);
/** **/
void INPUT_CNTL()
{
  FILE *fp;
	
  if( my_rank == 0 ){
    if( (fp=fopen("INPUT.DAT","r")) == NULL){
      fprintf(stdout,"input file cannot be opened!\n");
      exit(1);
    }
    fscanf(fp,"%s",HEADER);
    fscanf(fp,"%d",&ITER);
    fscanf(fp, "%lf %lf", &COND, &QVOL);
    fscanf(fp, "%lf",     &RESID);
    fclose(fp);
  }
  
  MPI_Bcast(HEADER ,80,MPI_CHAR,0,MPI_COMM_WORLD);
  MPI_Bcast(&ITER   , 1,MPI_INTEGER,0,MPI_COMM_WORLD);
  MPI_Bcast(&COND   , 1,MPI_DOUBLE,0,MPI_COMM_WORLD);
  MPI_Bcast(&QVOL   , 1,MPI_DOUBLE,0,MPI_COMM_WORLD);
  MPI_Bcast(&RESID  , 1,MPI_DOUBLE,0,MPI_COMM_WORLD);
  
  pfemRarray[0]= RESID;
  pfemIarray[0]= ITER;
}


